examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2218 @@
+    using rank2_type = typename ProblemDescription<dim>::rank2_type;
+
+    typedef std::array<dealii::LinearAlgebra::distributed::Vector<double>,
+                       problem_dimension>
+      vector_type;
+
+    TimeStep(const MPI_Comm &          mpi_communicator,
+             dealii::TimerOutput &     computing_timer,
+             const OfflineData<dim> &  offline_data,
+             const InitialValues<dim> &initial_values,
+             const std::string &       subsection = "TimeStep");
+
+    void prepare();
+
+    double step(vector_type &U, double t);
+
+  private:
+    const MPI_Comm &     mpi_communicator;
+    dealii::TimerOutput &computing_timer;
+
+    dealii::SmartPointer<const OfflineData<dim>>   offline_data;
+    dealii::SmartPointer<const InitialValues<dim>> initial_values;
+
+    dealii::SparseMatrix<double> dij_matrix;
+
+    vector_type temp;
+
+    double cfl_update;
+  };
+
+  // @sect4{Declaration of <code>SchlierenPostprocessor</code> class template}
+
+  // At its core, the Schilieren class implements the class member
+  // <code>compute_schlieren</code>. The main purpose of this class member
+  // is to compute auxiliary finite element field <code>schlieren</code>
+  // at each node, defined as
+  // \f[ \text{schlieren}[i] = e^{\beta \frac{ |\nabla r_i|
+  // - \min_j |\nabla r_j| }{\max_j |\nabla r_j| - \min_j |\nabla r_j| } } \f]
+  // where $r$ in principle could be any scalar finite element field.
+  // The natural candidate is choosing $r := \rho$. Schlieren postprocessing
+  // is a standard methodology to enhance the contrast of the visualization
+  // inspired in actual X-ray and shadowgraphy experimental techniques of
+  // visualization.
+
+  template <int dim>
+  class SchlierenPostprocessor : public dealii::ParameterAcceptor
+  {
+  public:
+    static constexpr unsigned int problem_dimension =
+      ProblemDescription<dim>::problem_dimension;
+
+    using rank1_type = typename ProblemDescription<dim>::rank1_type;
+
+    using vector_type =
+      std::array<dealii::LinearAlgebra::distributed::Vector<double>,
+                 problem_dimension>;
+
+    SchlierenPostprocessor(
+      const MPI_Comm &        mpi_communicator,
+      dealii::TimerOutput &   computing_timer,
+      const OfflineData<dim> &offline_data,
+      const std::string &     subsection = "SchlierenPostprocessor");
+
+    void prepare();
+
+    void compute_schlieren(const vector_type &U);
+
+    dealii::LinearAlgebra::distributed::Vector<double> schlieren;
+
+  private:
+    const MPI_Comm &     mpi_communicator;
+    dealii::TimerOutput &computing_timer;
+
+    dealii::SmartPointer<const OfflineData<dim>> offline_data;
+
+    dealii::Vector<double> r;
+
+    unsigned int schlieren_index;
+    double       schlieren_beta;
+  };
+
+  // @sect4{Declaration of <code>TimeLoop</code> class template}
+
+  // Placeholder here
+
+  template <int dim>
+  class TimeLoop : public dealii::ParameterAcceptor
```

Comment:
is this the main class? Maybe it makes sense to give it a different name? In particular, the difference between ``TimeLoop`` and ``TimeStep`` is not obvious to me.